Framework/Kernel/inc/MantidKernel/VMD.h

```diff
@@ -234,12 +234,12 @@ template <typename TYPE = double> class DLLExport VMDBase {
     boost::split(strs, str, boost::is_any_of(", "));
 
     std::vector<TYPE> vals;
-    for (size_t d = 0; d < strs.size(); d++) {
-      if (!strs[d].empty()) {
+    for (const auto &str : strs) {
+      if (!str.empty()) {
         TYPE v;
-        if (!Strings::convert(strs[d], v))
+        if (!Strings::convert(str, v))
           throw std::invalid_argument(
-              "VMDBase: Unable to convert the string '" + strs[d] +
+              "VMDBase: Unable to convert the string '" + str +
               "' to a number.");
         vals.push_back(v);
       }
```

Comment:
The entire function would look like this:
```
  VMDBase(const std::string &str) {
    StringTokenizer tokenizer(str, ", ", StringTokenizer::TOK_TRIM |
                                             StringTokenizer::TOK_IGNORE_EMPTY);

    std::vector<TYPE> vals;
    std::transform(tokenizer.cbegin(), tokenizer.cend(),
                   std::back_inserter(vals), [](const std::string &token) {
                     TYPE v;
                     if (!Strings::convert(token, v))
                       throw std::invalid_argument(
                           "VMDBase: Unable to convert the string '" + token +
                           "' to a number.");
                     return v;
                   });

    nd = vals.size();
    if (nd <= 0)
      throw std::invalid_argument("nd must be > 0");
    data = new TYPE[nd];

    std::copy(vals.cbegin(), vals.cend(), &data[0]);
  }
```
Not sure if you want to use the last line, but according to [this](http://en.cppreference.com/w/cpp/iterator/iterator_traits) you can always use raw pointers as iterators so it should be okay. I noticed `std::copy` could as well be used in the copy constructor etc.